src/middlewares/restrict-user-details.js

```diff
@@ -0,0 +1,102 @@
+module.exports = (config, { strapi }) => {
```

Comment:
Can you add a comment explaining why it's not enough to mark the fields private (under advanced https://docs.strapi.io/user-docs/content-type-builder/configuring-fields-content-type)?

Reply:
Done